cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/state/AppStateTable.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.store.state;
+
+import com.google.common.collect.ImmutableList;
+import io.cdap.cdap.spi.data.StructuredRow;
+import io.cdap.cdap.spi.data.StructuredTable;
+import io.cdap.cdap.spi.data.StructuredTableContext;
+import io.cdap.cdap.spi.data.table.field.Field;
+import io.cdap.cdap.spi.data.table.field.Fields;
+import io.cdap.cdap.spi.data.table.field.Range;
+import io.cdap.cdap.store.StoreDefinition;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Store for app state data.
+ * It does not wrap its operations in a transaction.
+ * It is up to the caller to decide what operations belong in a transaction.
+ */
+public class AppStateTable {
+  private final StructuredTable table;
+
+  public AppStateTable(StructuredTableContext context) {
+    this.table = context.getTable(StoreDefinition.AppStateStore.APP_STATE);
+  }
+
+  /**
+   * @param request with key fields {namespace, appName, stateKey} set
+   * @return state of application
+   * @throws IOException if state not found
+   */
+  public Optional<byte[]> getState(AppState request) throws IOException {
+    Optional<StructuredRow> row = table.read(getKeyFields(request));
+    return row.map(this::getStateValue);
+  }
+
+  /**
+   * @param request with all the fields set
+   * @throws IOException if exception while saving
+   */
+  public void saveState(AppState request) throws IOException {
+    table.upsert(getRow(request));
+  }
+
+  /**
+   * @param request with key fields {namespace, appName, stateKey} set
+   * @throws IOException if state not found
+   */
+  public void deleteState(AppState request) throws IOException {
```

Comment:
nit: 'State' is redundant in these method names, as they're all in the state table class. Can name them get/save/delete/deleteAll